x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinatorTests.java

```diff
@@ -2150,11 +2151,115 @@ void cleanFollowedRemoteIndices(ClusterState remoteClusterState, List<String> pa
         }
     }
 
+    public void testSystemIndicesAreNotAutoFollowed() {
+        ClusterState clusterState = null;
+        final int nbLeaderSystemIndices = randomIntBetween(1, 15);
+        for (int i = 0; i < nbLeaderSystemIndices; i++) {
+            String indexName = "." + i;
+            if (clusterState == null) {
+                clusterState = createRemoteClusterState(indexName, true, 0, true);
+            } else {
+                clusterState = createRemoteClusterState(clusterState, true, indexName);
+            }
+        }
+        final List<AutoFollowCoordinator.AutoFollowResult> autoFollowResults = executeAutoFollow(".*", clusterState);
+        assertThat(autoFollowResults.size(), equalTo(1));
+        assertThat(autoFollowResults.get(0).autoFollowExecutionResults, is(anEmptyMap()));
+    }
+
+    public void testSystemDataStreamsAreNotAutoFollowed() {
+        final List<AutoFollowCoordinator.AutoFollowResult> autoFollowResults =
+            executeAutoFollow("*.", createRemoteClusterStateWithDataStream(".test-data-stream"));
+
+        assertThat(autoFollowResults.size(), equalTo(1));
+        assertThat(autoFollowResults.get(0).autoFollowExecutionResults, is(anEmptyMap()));
+    }
+
+    private List<AutoFollowCoordinator.AutoFollowResult> executeAutoFollow(String indexPattern, final ClusterState finalRemoteState) {
```

Comment:
It would be nice to add a single test that also uses this method, that actually creates an index. It is sort of a test of the test, but given the complexity of this test method and the fact that if it just returns nothing, we are good, I think it is worthwhile.

Reply:
Added in https://github.com/elastic/elasticsearch/pull/72815/commits/d8a7c410ef5ea4a84f779cacc70006ef9336f4c0